src/gluonts/dataset/repository/_zenodo.py

```diff
@@ -0,0 +1,200 @@
+    "tourism_monthly": {
+        "file": "tourism_monthly_dataset.zip",
+        "record": "4656096",
+    },
+    "tourism_quarterly": {
+        "file": "tourism_quarterly_dataset.zip",
+        "record": "4656093",
+    },
+    "tourism_yearly": {
+        "file": "tourism_yearly_dataset.zip",
+        "record": "4656103",
+    },
+}
+
+
+def urllib_retrieve_hook(tqdm):
+    """Wraps tqdm instance.
+    Don'tqdm forget to close() or __exit__()
+    the tqdm instance once you're done with it (easiest using `with` syntax).
+    Example
+    -------
+    # >>> with tqdm(...) as tqdm:
+    # ...     reporthook = my_hook(tqdm)
+    # ...     urllib.urlretrieve(..., reporthook=reporthook)
+    """
+    last_b = [0]
+
+    def update_to(block=1, block_size=1, tsize=None):
+        """
+        block  : int, optional
+            Number of blocks transferred so far [default: 1].
+        block_size  : int, optional
+            Size of each block (in tqdm units) [default: 1].
+        tsize  : int, optional
+            Total size (in tqdm units). If [default: None] remains unchanged.
+        """
+        if tsize is not None:
+            tqdm.total = tsize
+        tqdm.update((block - last_b[0]) * block_size)
+        last_b[0] = block
+
+    return update_to
+
+
+def download_dataset(description: Dict[str, str], path: Path):
+    file = description["file"]
+    with tqdm(
+        [],
+        unit="B",
+        unit_scale=True,
+        unit_divisor=1024,
+        miniters=5,
+        desc=f"download {file}",
+    ) as _tqdm:
+        request.urlretrieve(
+            f"{ROOT}/{description['record']}/files/{file}",
+            filename=str(path / description["file"]),
+            reporthook=urllib_retrieve_hook(_tqdm),
+        )
+
+
+def frequency_converter(freq: str):
+    parts = freq.split("_")
+    if len(parts) == 1:
+        return freq[0].upper()
+    if len(parts) == 2 and parts[0].isnumeric():
+        return f"{parts[0]}{parts[1].upper()}"
+
+
+def save_metadata(
+    dataset_path: Path, cardinality: int, freq: str, prediction_length: int
+):
+    with open(dataset_path / "metadata.json", "w") as file:
+        json.dump(
+            metadata(
+                cardinality=cardinality,
+                freq=freq,
+                prediction_length=prediction_length,
+            ),
+            file,
+        )
+
+
+def save_datasets(path: Path, data: List[Dict], train_offset: int):
+    train = path / "train"
+    train.mkdir(exist_ok=True)
+    test = path / "test"
+    test.mkdir(exist_ok=True)
+
+    with open(train / "data.json", "wb") as train_fp, open(
+        test / "data.json", "wb"
+    ) as test_fp:
+        for data_entry in tqdm(
+            data, total=len(data), desc="creating json files"
+        ):
+            dic = to_dict(
+                target_values=data_entry["target"],
+                start=str(data_entry["start_timestamp"]),
+            )
+
+            test_fp.write(json.dumps(dic).encode("utf-8"))
```

Comment:
I think you can use `dataset.jsonl.dump` here

https://github.com/awslabs/gluon-ts/blob/6d2dc64a6eba276bebdd74da4d5b7081ae13e4a2/src/gluonts/dataset/jsonl.py#L29-L31

Reply:
I think when we want to use it, it would look something like:

```
    train_data = []
    test_data = []
    for data_entry in tqdm(
        data, total=len(data), desc="creating json files"
    ):
        dic = to_dict(
            target_values=data_entry["target"],
            start=str(data_entry["start_timestamp"]),
        )
        test_data.append(dic) 
        dic["target"] = dic["target"][:-train_offset]
        train_data.append(dic) 
    with open(train / "data.json", "w") as train_fp, open(
        test / "data.json", "w"
    ) as test_fp:
        dump(train_fp, train_data)
        dump(test_fp, test_data)
```
Because `dataset.jsonl.dump` iterates over the objects parameter. Therefore, I would suggest to just leave it as it. What do you think?

Reply:
implemented